splitio/client/factory.py

```diff
@@ -352,24 +346,47 @@ def _build_in_memory_factory(api_key, cfg, sdk_url=None, events_url=None,  # pyl
         EventSynchronizer(apis['events'], storages['events'], cfg['eventsBulkSize']),
         ImpressionsCountSynchronizer(apis['impressions'], imp_manager),
     )
-    imp_count_sync_task = ImpressionsCountSyncTask(synchronizers.impressions_count_sync.synchronize_counters) if cfg['impressionsMode'] == 'OPTIMIZED' else None
 
-    tasks = SplitTasks(
-        SplitSynchronizationTask(
-            synchronizers.split_sync.synchronize_splits,
-            cfg['featuresRefreshRate'],
-        ),
-        SegmentSynchronizationTask(
-            synchronizers.segment_sync.synchronize_segments,
-            cfg['segmentsRefreshRate'],
-        ),
-        ImpressionsSyncTask(
-            synchronizers.impressions_sync.synchronize_impressions,
-            cfg['impressionsRefreshRate'],
-        ),
-        EventsSyncTask(synchronizers.events_sync.synchronize_events, cfg['eventsPushRate']),
-        ImpressionsCountSyncTask(synchronizers.impressions_count_sync.synchronize_counters),
-    )
+    if cfg['impressionsMode'] == ImpressionsMode.NONE:
```

Comment:
splitTask = SplitSynchronizationTask(synchronizers.split_sync.synchronize_splits, cfg['featuresRefreshRate'])
segmentTask = ...
impressionTask = ...
eventTask = ...
impressionsCountTask = None
uniqueKeysTask = None
clearFilterTask = None

if cfg['impressionsMode'] == ImpressionsMode.NONE:
    synchronizers.set_none_sync(
        UniqueKeysSynchronizer(InMemorySenderAdapter(apis['telemetry']), imp_strategy._unique_keys_tracker),
        ClearFilterSynchronizer(imp_strategy._unique_keys_tracker)
    )
    impressionsCountTask = ImpressionsCountSyncTask(synchronizers.impressions_count_sync.synchronize_counters)
    uniqueKeysTask = UniqueKeysSyncTask(synchronizers.unique_keys_sync.SendAll)
    clearFilterTask = ClearFilterSyncTask(synchronizers.clear_filter_sync.clearAll)

tasks = SplitTasks(splitTask, segmentTask, ...)

Reply:
you can do the same for synchronizer object and avoid the method `set_none_sync`

Reply:
I had that initially, for some reason the unique_keys_sync object was messed up always get an exception object does no have "SendAll" method, only when defining the object directly in synchronizer[] it worked.